src/main/scala/org/bitcoins/core/gen/ScriptGenerators.scala

```diff
@@ -73,36 +103,64 @@ trait ScriptGenerators extends BitcoinSLogger {
     multiSignatureScriptPubKey
   }
 
+  //TODO: Improve the format of each scriptPubKey generator to return a privateKey needed to spend it.
   def p2shScriptPubKey : Gen[P2SHScriptPubKey] = for {
     randomScriptPubKey <- pickRandomNonP2SHScriptPubKey
   } yield P2SHScriptPubKey(randomScriptPubKey)
 
-
   def emptyScriptPubKey = p2pkScriptPubKey.map(_ => EmptyScriptPubKey)
 
   def pickRandomNonP2SHScriptPubKey : Gen[ScriptPubKey] = {
-    val randomNum = (scala.util.Random.nextInt() % 3).abs
+    val randomNum = (scala.util.Random.nextInt() % 5).abs
     if (randomNum == 0) p2pkScriptPubKey
     else if (randomNum == 1) p2pkhScriptPubKey
+    else if (randomNum == 2) cltvScriptPubKey
+    else if (randomNum == 3) csvScriptPubKey
     else multiSigScriptPubKey
   }
 
+  def pickRandomNonLockTimeScriptPubKey(pubKey : ECPublicKey) : Gen[ScriptPubKey] = {
+    val randomNum = (scala.util.Random.nextInt() % 3).abs
+    if (randomNum == 0) P2PKScriptPubKey(pubKey)
+    else if (randomNum == 1) P2PKHScriptPubKey(pubKey)
+    else P2PKHScriptPubKey(pubKey)
+  }
+
+  def pickRandomNonLockTimeScriptPubKey : Gen[(ScriptPubKey, Seq[ECPrivateKey])] = {
```

Comment:
Why is there two different instances of this function, one that takes a `pubKey: ECPublicKey` and one that doesn't?